src/Gate.jl

```diff
@@ -206,6 +206,15 @@ abstract type Control{Op<:Operator} <: Operator{NamedTuple{(:target,),Tuple{Oper
 Base.length(::Type{Control{T}}) where {T<:Operator} = 1 + length(T)
 parameters(::Type{Control{Op}}) where {Op} = parameters(Op)
 
+"""
+    SU(N)(lane_1, lane_2, ..., lane_log2(N))
+
+A multi-qubit random unitary operator that acts on `log2(N)` qubits.
+"""
+abstract type SU{N} <: Operator{NamedTuple{(:values, :N), Tuple{Matrix, Int}}} where {T, N<:Int} end
```

Comment:
Where is `T` used? `N` is not a parameter but a type-parameter of `SU`.
```suggestion
abstract type SU{N} <: Operator{NamedTuple{(:values,), Tuple{Matrix}}} end
```